src/System.Net.WebSockets.Client/tests/ClientWebSocketTest.cs

```diff
@@ -27,9 +28,46 @@ public ClientWebSocketTest(ITestOutputHelper output)
             _output = output;
         }
 
+        public static IEnumerable<object[]> UnavailableWebSocketServers
+        {
+            get
+            {
+                Uri server;
+                
+                // Unknown server.
+                {
+                    server = new Uri(string.Format("ws://{0}", Guid.NewGuid().ToString()));
+                    yield return new object[] { server };
+                }
+
+                // Known server but not a real websocket endpoint.
+                {
+                    server = HttpTestServers.RemoteEchoServer;
+                    var ub = new UriBuilder("ws", server.Host, server.Port, server.PathAndQuery);
+
+                    yield return new object[] { ub.Uri };
+                }
+            }
+        }
+
         private static bool WebSocketsSupported { get { return WebSocketHelper.WebSocketsSupported; } }
 
 #region Connect
+        [ConditionalTheory("WebSocketsSupported"), MemberData("UnavailableWebSocketServers")]
```

Comment:
Will these actually work for NetNative or Desktop? Since the tests are now different from the ones we're running for Desktop sign-off and so may diverge and ensure different behaviors.

Until there's some common CoreFX/NetNative/Desktop, I'd prefer having all tests as close as possible in implementation especially because this component is not yet finished.